tst/com/amazon/corretto/crypto/provider/test/AesKeyWrapTest.java

```diff
@@ -0,0 +1,550 @@
+    assertThrows(IllegalStateException.class, () -> c.update(new byte[8], 0, 1));
+  }
+
+  @Test
+  public void testBadInputs_doFinal() throws Exception {
+    final Cipher c = getCipher(TestUtil.NATIVE_PROVIDER);
+    assertThrows(IllegalStateException.class, () -> c.doFinal(new byte[8]));
+    c.init(Cipher.WRAP_MODE, getAesKey(128 / 8));
+    assertThrows(IllegalStateException.class, () -> c.doFinal(new byte[8]));
+    c.init(Cipher.UNWRAP_MODE, getAesKey(128 / 8));
+    assertThrows(IllegalStateException.class, () -> c.doFinal(new byte[8]));
+    c.init(Cipher.ENCRYPT_MODE, getAesKey(128 / 8));
+    final int dataSize = 1024;
+    c.update(new byte[dataSize]);
+    assertThrows(
+        ShortBufferException.class, () -> c.doFinal(new byte[8], 0, 1, new byte[dataSize + 1], 0));
+  }
+
+  @Test
+  public void testBadInputs_wrapAndUnwrap() throws Exception {
+    final Key key = getAesKey(128 / 8); // NOTE: use for both wrapping and being wrapped
+    final Cipher c = getCipher(TestUtil.NATIVE_PROVIDER);
+
+    // Bad modes
+    c.init(Cipher.UNWRAP_MODE, key);
+    assertThrows(IllegalStateException.class, () -> c.wrap(key));
+    c.init(Cipher.ENCRYPT_MODE, key);
+    assertThrows(IllegalStateException.class, () -> c.wrap(key));
+    c.init(Cipher.DECRYPT_MODE, key);
+    assertThrows(IllegalStateException.class, () -> c.wrap(key));
+    c.init(Cipher.WRAP_MODE, key);
+    final byte[] wrapped = c.wrap(key);
+    assertThrows(IllegalStateException.class, () -> c.unwrap(wrapped, "AES", Cipher.SECRET_KEY));
+    c.init(Cipher.ENCRYPT_MODE, key);
+    assertThrows(IllegalStateException.class, () -> c.unwrap(wrapped, "AES", Cipher.SECRET_KEY));
+    c.init(Cipher.DECRYPT_MODE, key);
+    assertThrows(IllegalStateException.class, () -> c.unwrap(wrapped, "AES", Cipher.SECRET_KEY));
+
+    // Bad key
+    c.init(Cipher.WRAP_MODE, key);
+    assertThrows(InvalidKeyException.class, () -> c.wrap(NULL_KEY));
+  }
+
+  @Test
+  public void testBadInputs_wrongUnwrapKey() throws Exception {
+    final Cipher c = getCipher(TestUtil.NATIVE_PROVIDER);
+    final SecretKey key1 = getAesKey(16);
+    final SecretKey key2 = getAesKey(16);
+    final SecretKey secret = new SecretKeySpec(TestUtil.getRandomBytes(32), "Generic");
+
+    c.init(Cipher.WRAP_MODE, key1);
+    final byte[] wrappedKey = c.wrap(secret);
+
+    c.init(Cipher.UNWRAP_MODE, key2);
+    assertThrows(
+        InvalidKeyException.class, () -> c.unwrap(wrappedKey, "Generic", Cipher.SECRET_KEY));
+
+    c.init(Cipher.UNWRAP_MODE, key1);
+    final Key unwrappedKey = c.unwrap(wrappedKey, "Generic", Cipher.SECRET_KEY);
+    assertArraysHexEquals(secret.getEncoded(), unwrappedKey.getEncoded());
+  }
+
+  @Test
+  public void threadStorm() throws GeneralSecurityException, InterruptedException {
+    final byte[] rngSeed = TestUtil.getRandomBytes(20);
+    System.out.println("RNG Seed: " + Arrays.toString(rngSeed));
```

Comment:
Let's drop this line. At some point we need to get rid of all these printlns from the test code, but I'd like to prevent adding more of them.